components/remote_settings/src/client.rs

```diff
@@ -271,10 +281,217 @@ struct AttachmentsCapability {
+    }
+
+    /// Sets an option to only return items whose `field` is a string that
+    /// contains the substring `value`. `value` can contain `*` wildcards.
+    pub fn like(&mut self, field: impl Into<String>, value: impl Into<String>) -> &mut Self {
+        self.filters.push(Filter::Like(field.into(), value.into()));
+        self
+    }
+
+    /// Sets an option to only return items that have the given `field`.
+    pub fn has(&mut self, field: impl Into<String>) -> &mut Self {
+        self.filters.push(Filter::Has(field.into()));
+        self
+    }
+
+    /// Sets an option to only return items that do not have the given `field`.
+    pub fn has_not(&mut self, field: impl Into<String>) -> &mut Self {
+        self.filters.push(Filter::HasNot(field.into()));
+        self
+    }
+
+    /// Sets an option to return items in `order` for the given `field`.
+    pub fn sort(&mut self, field: impl Into<String>, order: SortOrder) -> &mut Self {
+        self.sort.push(Sort(field.into(), order));
+        self
+    }
+
+    /// Sets an option to only return the given `field` of each item.
+    ///
+    /// The special `id` and `last_modified` fields are always returned.
+    pub fn field(&mut self, field: impl Into<String>) -> &mut Self {
+        self.fields.push(field.into());
+        self
+    }
+
+    /// Sets the option to return at most `count` items.
+    pub fn limit(&mut self, count: u64) -> &mut Self {
+        self.limit = Some(count);
+        self
+    }
+
+    /// Returns an iterator of (name, value) query pairs for these options.
+    pub fn iter_query_pairs(&self) -> impl Iterator<Item = (Cow<str>, Cow<str>)> {
+        self.filters
+            .iter()
+            .map(Filter::as_query_pair)
+            .chain({
+                // For sorting (https://docs.kinto-storage.org/en/latest/api/1.x/sorting.html),
+                // the query pair syntax is `_sort=field1,-field2`, where the
+                // fields to sort by are specified in a comma-separated ordered
+                // list, and `-` indicates descending order.
+                if self.sort.is_empty() {
+                    None
+                } else {
+                    Some((
+                        "_sort".into(),
+                        (self
+                            .sort
+                            .iter()
+                            .map(Sort::as_query_value)
+                            .collect::<Vec<_>>()
+                            .join(","))
+                        .into(),
+                    ))
+                }
+            })
+            .chain({
+                // For selecting fields (https://docs.kinto-storage.org/en/latest/api/1.x/selecting_fields.html),
+                // the query pair syntax is `_fields=field1,field2`.
+                if self.fields.is_empty() {
+                    None
+                } else {
+                    Some(("_fields".into(), self.fields.join(",").into()))
+                }
```

Comment:
The newish `bool` method can be nice in places like this (and above)
```suggestion
                !(self.fields.is_empty())
                    .then_some(("_fields".into(), self.fields.join(",").into()))
```

Reply:
Thanks @pjenvey! 🏆 I actually tried writing it with `.then(...)` first, like:

```rs
.chain(
    (!self.fields.is_empty()).then(|| ("_fields".into(), self.fields.join(",").into())),
)
```

...But that felt a little cramped, and I thought the extra parentheses around `(!self.fields.is_empty())` made it easy to miss the `!`, so I wrote it out explicitly. Although, with the comment and block there, maybe it's not too bad?

```rs
.chain({
    // For selecting fields (https://docs.kinto-storage.org/en/latest/api/1.x/selecting_fields.html),
    // the query pair syntax is `_fields=field1,field2`.
    (!self.fields.is_empty()).then(|| ("_fields".into(), self.fields.join(",").into()))
})
```

WDYT?

Reply:
It's just a suggestion so feel free to ignore. I'll just note that I've used this a few times chaining onto the same `!is_empty()` method call and it already feels like a pretty common, easily identifiable pattern to me: if there's something in this collection, use it.

Reply:
Perfect, thanks—it helps to know that you've used this a few times, and that it feels like a common pattern! I folded in your suggestion 😁